src/app/pages/registration/registration-page.component.ts

```diff
@@ -26,40 +23,27 @@ export class RegistrationPageComponent implements OnInit {
   error$: Observable<HttpError>;
 
   constructor(
-    private accountFacade: AccountFacade,
     private route: ActivatedRoute,
-    private featureToggle: FeatureToggleService,
     private registrationFormConfiguration: RegistrationFormConfigurationService
   ) {}
 
   submitted = false;
 
   registrationConfig: RegistrationConfigType;
 
-  fields$: Observable<FormlyFieldConfig[]>;
+  fields: FormlyFieldConfig[];
   model: Record<string, unknown>;
   options: FormlyFormOptions;
   form = new FormGroup({});
 
   ngOnInit() {
-    this.error$ = merge(this.accountFacade.userError$, this.accountFacade.ssoRegistrationError$);
-    this.fields$ = this.route.queryParamMap.pipe(
-      tap((paramMap: ParamMap) => (this.model = this.processParamMap(paramMap))),
-      map((paramMap: ParamMap) => ({
-        sso: paramMap.get('sso') === 'true',
-        userId: paramMap.get('userid'),
-        businessCustomer: this.featureToggle.enabled('businessCustomerRegistration'),
-      })),
-      tap(config => (this.registrationConfig = config)),
-      tap(
-        config =>
-          (this.options = this.registrationFormConfiguration.getRegistrationFormConfigurationOptions(
-            config,
-            this.model
-          ))
-      ),
-      map(config => this.registrationFormConfiguration.getRegistrationFormConfiguration(config))
-    );
+    this.error$ = this.registrationFormConfiguration.getErrorSources();
+
+    const snapshot = this.route.snapshot;
+    this.model = this.registrationFormConfiguration.extractModel(snapshot);
+    this.registrationConfig = this.registrationFormConfiguration.extractConfig(snapshot);
+    this.options = this.registrationFormConfiguration.getOptions(this.registrationConfig, this.model);
+    this.fields = this.registrationFormConfiguration.getRegistrationFormConfiguration(this.registrationConfig);
```

Comment:
Somehow this is still bothering me. It's only naming! But I would prefer to read this line like this:
```
this.fields = this.registrationFormConfiguration.getFields(this.registrationConfig);
```